### ISSUE TITLE
Exception retrieving untagged images on api >= 1.24

### ISSUE THREAD

**Issue by hobosteaux** [docker/docker-py#1351]
Docker API >= 1.24 will return a null object if image tags DNE instead of not including it in the response. This makes the dict.get fail to catch the null case and the list comprehension to iterate over a non-iterable.

```
  File "<stdin>", line 1, in <module>
  File "docker/models/images.py", line 16, in __repr__
    return "<%s: '%s'>" % (self.__class__.__name__, "', '".join(self.tags))
  File "docker/models/images.py", line 34, in tags
    tag for tag in self.attrs.get('RepoTags', [])
TypeError: 'NoneType' object is not iterable
```

This is similar to an issue seen in [salt](https://github.com/saltstack/salt/pull/35447/commits/b833b5f9587534d3b843a026ef91abc4ec929d0f)

Was able to get things working with a pretty quick change:
```
diff --git a/docker/models/images.py b/docker/models/images.py
index 32068e6..39a640d 100644
--- a/docker/models/images.py
+++ b/docker/models/images.py
@@ -30,9 +30,11 @@ class Image(Model):
         """
         The image's tags.
         """
+        tags = self.attrs.get('RepoTags', [])
+        if tags is None:
+            return []
         return [
-            tag for tag in self.attrs.get('RepoTags', [])
-            if tag != '<none>:<none>'
+            tag for tag in tags if tag != '<none>:<none>'
         ]
 
     def history(self):
```

**Comment by shin-** [comment]
Thanks for the report, and the patch!

**Comment by bugchecker** [comment]
just enough
```
diff --git a/docker/models/images.py b/docker/models/images.py
index 32068e6..21139ae 100644
--- a/docker/models/images.py
+++ b/docker/models/images.py
@@ -31,7 +31,7 @@ def tags(self):
         The image's tags.
         """
         return [
-            tag for tag in self.attrs.get('RepoTags', [])
+            tag for tag in (self.attrs.get('RepoTags') or [])
             if tag != '<none>:<none>'
         ]
```

**Comment by aebm** [comment]
Hi this issue stills happens in

```
docker==2.0.1
docker-pycreds==0.2.1
Python 2.7.6
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:26:30 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:26:30 2017
 OS/Arch:      linux/amd64
```

To reproduce this issue you first need to pull have an image without tags. For example pulling an image by hash
```bash
docker pull busybox@sha256:${IMAGE_HASH}
```

Then run this snippet 

```python
#!/usr/bin/env python

import docker
client = docker.from_env()
image_list = client.images.list(all=True)
for image in image_list:
    print 'Id:', image.short_id
    print ' Tags:', ','.join(image.tags)
```

You get this exception
```
Traceback (most recent call last):
  File "./xxxx.py", line 8, in <module>
    print ' Tags:', ','.join(image.tags)
  File "x/python2.7/site-packages/docker/models/images.py", line 34, in tags
    tag for tag in self.attrs.get('RepoTags', [])
TypeError: 'NoneType' object is not iterable

**Comment by sean-abbott** [comment]
Any chance of this fix getting back-ported to the 1.10.x branch? Pretty much all of tools I use that use this library are still on that branch...
